hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+          HoodieCLI.tableMetadata.getHadoopConf()).globStatus(new Path(logFilePathPattern)))
+          .map(status -> status.getPath().toString()).collect(Collectors.toList());
+
+    // TODO : readerSchema can change across blocks/log files, fix this inside Scanner
+    AvroSchemaConverter converter = new AvroSchemaConverter();
+    Schema readerSchema = converter.convert(readSchemaFromLogFile(new Path(logFilePaths.get(0))));
+
+      List<IndexedRecord> allRecords = new ArrayList<>();
+
+      if(shouldMerge) {
+        System.out.println("===========================> MERGING RECORDS <===================");
+        HoodieCompactedLogRecordScanner scanner = new HoodieCompactedLogRecordScanner(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+            HoodieCLI.tableMetadata.getHadoopConf()),
+            HoodieCLI.tableMetadata.getBasePath(), logFilePaths, readerSchema,
+            HoodieCLI.tableMetadata.getActiveTimeline().getCommitTimeline().lastInstant().get().getTimestamp(),
+            Long.valueOf(HoodieCompactionConfig.DEFAULT_MAX_SIZE_IN_MEMORY_PER_COMPACTION_IN_BYTES));
+        for (HoodieRecord<? extends HoodieRecordPayload> hoodieRecord : scanner) {
+          Optional<IndexedRecord> record = hoodieRecord.getData().getInsertValue(readerSchema);
+          if (allRecords.size() >= limit) {
+            break;
+          }
+          allRecords.add(record.get());
+        }
+      } else {
+
+        for (String logFile : logFilePaths) {
+          Schema writerSchema = new AvroSchemaConverter().convert(readSchemaFromLogFile(new Path(logFile)));
+          HoodieLogFormat.Reader reader = HoodieLogFormat.newReader(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+              HoodieCLI.tableMetadata.getHadoopConf()),
+              new HoodieLogFile(new Path(logFile)), writerSchema, true);
+          // read the avro blocks
+          while (reader.hasNext()) {
+            HoodieLogBlock n = reader.next();
+            if(n instanceof HoodieAvroDataBlock) {
+              HoodieAvroDataBlock blk = (HoodieAvroDataBlock) n;
+              List<IndexedRecord> records = blk.getRecords();
+              allRecords.addAll(records);
+              if(allRecords.size() >= limit) {
+                break;
+              }
+            }
+          }
+          if(allRecords.size() >= limit) {
+            break;
+          }
+        }
+      }
+      String [][] rows = new String[allRecords.size() + 1][];
+      int i = 0;
+      for(IndexedRecord record : allRecords) {
+        String [] data = new String[1];
+        data[0] = record.toString();
+        rows[i] = data;
+        i++;
+      }
+      return HoodiePrintHelper.print(
+          new String[]{"Records"}, rows);
+  }
+
+  /**
+   * Read the schema from the log file on path
+   */
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private MessageType readSchemaFromLogFile(Path path) throws IOException {
```

Comment:
can this belong in a utils class in hoodie-common?

Reply:
nit: rename to `readLatestSchemaFromLogFile`

Reply:
done